tss-esapi/src/utils/mod.rs

```diff
@@ -124,7 +131,7 @@ pub fn create_restricted_decryption_rsa_public(
             )
             .build()?,
         )
-        .with_rsa_unique_identifier(&PublicKeyRsa::new_empty_with_size(rsa_key_bits))
+        .with_rsa_unique_identifier(&PublicKeyRsa::default())
```

Comment:
Is this change a necessary part of the fix, or just some tidying up? I assume it's intentional that we no longer take account of the key size in this line?

Reply:
I'm doubting myself now, let me check again if this had an effect on the primary key.

The short answer is that this has no effect - the `unique` field is replaced by the TPM during the operation anyway, so it shouldn't matter what we put there. As long as the content is in line with what the type of the object is, it should be ok. The old approach (obviously) put a bunch of zero bytes in there and set the length to that of the RSA key, the new one just sets it to an empty buffer. I changed it because I thought it might be influencing the seeding of the key. Let me check again.

Reply:
Yeah, so it seems `ObjectChangeAuth` fails because it doesn't recognise the primary key as the "true" parent of the keys loaded from contexts. That means the "name" of the primary key is computed over the `TPM2B_PUBLIC` received from the user (or, in any case, includes this `unique` field), and since before we were simply leaving the field empty, adding these empty bytes would prevent us from migrating in this way.